### PR TITLE
[WIP]Upgrade Netty to 4.1.17

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

Submitted this PR to see if there is anything broken.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by ash211** [comment]
Looks like a fix for https://issues.apache.org/jira/browse/SPARK-19552 -- should that be reopened now that netty is deprecating 4.0.x so we can't do it "Later" anymore?

**Comment by zsxwing** [comment]
@ash211 Good call. I will reopen it.

File: common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipher.java [hunk 1]
```diff
@@ -203,6 +203,63 @@ public long transfered() {
       return transferred;
     }
 
+    @Override
```
Comment (vanzin) [inline common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipher.java]: I prototyped this change internally for a different reason, and I chose to create an abstract class for all this boilerplate, which seems cleaner to me.

```
public abstract class AbstractFileRegion extends AbstractReferenceCounted implements FileRegion {

  @Override
  @SuppressWarnings("deprecation")
  public final long transfered() {
    return transferred();
  }

  @Override
  public FileRegion retain() {
    super.retain();
    return this;
  }

  @Override
  public FileRegion retain(int increment) {
    super.retain(increment);
    return this;
  }

  @Override
  public FileRegion touch() {
    return touch(null);
  }

  @Override
  public FileRegion touch(Object hint) {
    return this;
  }

}
```

Comment (zsxwing) [reply common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipher.java]: Good point! Updated!

**Pull request closed without merging**